server/data_provider_canvas_esb.rb

```diff
@@ -124,38 +124,48 @@ def canvasAPICalendarEventsURL(uniqname, canvas_courses)
+    all_courses = canvas_courses.inject([]) do |all, course|
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: canvas course array check course: [#{course}]"
+      all.concat(canvasCalendarEventsSingleCourse(course, uniqname))
+      # This must be returned to be used as the accumulator for inject.
+      all
+    end
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: calendar_events_array count: #{all_courses.length} all_courses: #{all_courses}"
 
-    calendar_events_url = canvasAPICalendarEventsURL(uniqname, canvas_courses)
+    WAPIResultWrapper.new(WAPI::SUCCESS, "got calendar_events from canvas esb", all_courses)
+  end
 
-    r = @canvasESB_w.get_request calendar_events_url
+  # Get calendar events for this single course (so can handle error conditions on a per course basis).
+  def canvasCalendarEventsSingleCourse(canvas_course, uniqname)
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: get calendar events for this course: >>#{canvas_course}<<"
+    calendar_events_url = canvasAPICalendarEventsURL(uniqname, [canvas_course])
 
+    r = @canvasESB_w.get_request calendar_events_url
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: canvas_course >>#{canvas_course}<< r:#{r.inspect}"
     canvas_body = r.result
-    canvas_body_ruby = JSON.parse canvas_body
-
-    return WAPIResultWrapper.new(WAPI::SUCCESS, "got calendar_events from canvas esb", canvas_body_ruby)
-  end
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: canvas_course >>#{canvas_course}<< canvas_body:#{canvas_body.inspect}"
 
+    # if the course access is unauthorized then ignore the course.  This can happen if the student status in the course is inactive.
+    p = Regexp.new(/401 Unauthorized/)
+    if canvas_body.class == RestClient::Unauthorized
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: canvas_course >>#{canvas_course}<< UNAUTHORIZED<<<<<<<<<"
+      return []
+    end
 
-  #  <canvas server>/api/v1/users/self/upcoming_events
-  # actually call out to canvas and return the value.  Caller will reformat if necessary.
-  # def canvasESBToDoLMS_upcoming_events(uniqname, security_file, esb_application)
-  #   logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: ############### call canvas ESB todolms esb_application: #{esb_application}"
-  #   logger.debug "##{self.class.to_s}:#{__method__}:#{__LINE__}: canvas ESB: @canvasESB_w: [#{@canvasESB_w}]"
-  #
-  #   r = @canvasESB_w.get_request "/users/self/upcoming_events?as_user_id=sis_login_id:#{uniqname}"
-  #
-  #   canvas_body = r.result
-  #   canvas_body_ruby = JSON.parse canvas_body
-  #
-  #   return WAPIResultWrapper.new(WAPI::SUCCESS, "got todos from canvas esb", canvas_body_ruby)
-  # end
-
-  # fake list of "canvas_courses"=>["43412", "44525", "44526", "44631", "44630", "44528", "44530"]}
+    canvas_body_ruby = JSON.parse canvas_body
```

Comment:
I was confused by the name `canvas_body_ruby`.  After I've verified what I thought `JSON.parse` is doing, I think a better name would be `canvas_body_objects`.  Likewise, maybe `canvas_body` should be `canvas_body_json`.